editor/src/core/shared/atom-with-pub-sub.tsx

```diff
@@ -17,6 +26,16 @@ export function atomWithPubSub<T>(options: { key: string; defaultValue: T }): At
   const newAtom: AtomWithPubSub<T> = {
     key: key,
     currentValue: defaultValue,
+    Provider: ({ children, value }) => {
+      const updateAtomSynchronously = usePubSubAtomWriteOnlyInner(newAtom, 'sync')
+      newAtom.currentValue = value // TODO this is sneaky and we should use API instead
+      useIsomorphicLayoutEffect(() => {
+        unstable_batchedUpdates(() => {
```

Comment:
We might be able to bin this off with React 18 :crossed_fingers: https://github.com/reactwg/react-18/discussions/21